dcos_launch/gce.py

```diff
@@ -0,0 +1,87 @@
+import os
+import logging
+import dcos_launch.util
+from dcos_test_utils import gce
+from retrying import retry
+
+log = logging.getLogger(__name__)
+
+
+class BareClusterLauncher(dcos_launch.util.AbstractLauncher):
+    # Launches a homogeneous cluster of plain GMIs intended for onprem DC/OS
+    def __init__(self, config):
+        # config = get_hardcoded_basic_config(os_name, instance_count)
+        self.cwd = os.path.dirname(os.path.realpath(__file__))
+        gce_folder = os.path.expanduser('~') + '/.gce'
+        credentials_path = gce_folder + '/credentials.json'
+        self.keys_path = gce_folder + '/keys'
+        self.gce_wrapper = gce.GceWrapper(credentials_path)
+        self.private_key_path = self.keys_path + '/' + config['deployment_name'] + '.priv'
+        self.public_key_path = self.keys_path + '/' + config['deployment_name'] + '.pub'
+
+        template_path = config.get('template_path', os.path.join(self.cwd, 'templates', 'gce-template.yaml'))
+        with open(template_path, 'r') as tfile:
+            template_content = tfile.read()
+        template_content = template_content.replace('${network}', config['network'])
+        template_content = template_content.replace('${sourceImage}', config['sourceImage'])
+        template_content = template_content.replace('${machineType}', config['machineType'])
+        template_content = template_content.replace('${zone}', config['zone'])
+        template_content = template_content.replace('${project_id}', self.gce_wrapper.project_id)
+        config['template_content'] = template_content
+
+        self.config = config
+        self.deployment = gce.Deployment(self.gce_wrapper, config['deployment_name'], config['zone'])
+
+    def create(self):
+        response = self.gce_wrapper.deploy_instances(self.config)
+
+        if self.config['key_helper']:
```

Comment:
It is guaranteed to exist. The philosophy I am going with here is that launch.config figures out all this stuff first so that way this module is only concerned with the actual launching, not user arguments